pkg/util/framer/framer.go

```diff
@@ -131,13 +131,13 @@ func NewJSONFramedReader(r io.ReadCloser) io.ReadCloser {
 func (r *jsonFrameReader) Read(data []byte) (int, error) {
 	// Return whatever remaining data exists from an in progress frame
 	if n := len(r.remaining); n > 0 {
-		if n <= cap(data) {
+		if n <= len(data) {
```

Comment:
You have to pass a buffer with len(data) > 0 to read - otherwise you read nothing.  The slice is tricky here, if data was an array it would be clearer.  Read treats data like an array, not a slice.